src/gui/tray/notificationhandler.cpp

```diff
@@ -159,12 +159,19 @@ void ServerNotificationHandler::slotNotificationsReceived(const QJsonDocument &j
 
         // Add another action to dismiss notification on server
         // https://github.com/owncloud/notifications/blob/master/docs/ocs-endpoint-v1.md#deleting-a-notification-for-a-user
-        ActivityLink al;
-        al._label = tr("Dismiss");
-        al._link = Utility::concatUrlPath(ai->account()->url(), notificationsPath + "/" + QString::number(a._id)).toString();
-        al._verb = "DELETE";
-        al._primary = false;
-        a._links.append(al);
+        const auto deleteVerb = "DELETE";
+        const auto deleteAction = std::find_if(a._links.begin(), a._links.end(), [deleteVerb](const ActivityLink& link){
```

Comment:
you are mixing `std::cend(a._links)` and calls to `a._links.end()`
better use always `std::cend` or `std::begin`

Reply:
@camilasan You could make it even shorter:
`if (std::find_if(std::cbegin(a._links), std::cend(a._links), [](const ActivityLink& link) { return link._verb == QStringLiteral("DELETE"); }) == std::cend(a._links)) {}`

Reply:
> better use always std::cend or std::begin

@mgallien I have to ask why?

Reply:
will be easier to read if we use the same way to get the `begin()` and `end()` iterators in all the code
so either way but always the same